modules/ting/src/Search/BooleanStatementGroup.php

```diff
@@ -67,7 +67,7 @@ public function setLogicOperator($logic_operator) {
    * @return TingSearchFieldFilter[]
    *   One or more instances of TingSearchFieldFilter.
    */
-  public function getStatements() {
+  public function &getStatements() {
```

Comment:
Passing the statements by reference is not a good idea. We do not want our search result to mutate.  If we need to modify the filters on a search request then we should retrieve the filters, modify them and then create a new search request using `withFilters()`.

Reply:
Resolved.